code/datums/components/material_container.dm

```diff
@@ -52,7 +52,7 @@
 
 /datum/component/material_container/proc/OnAttackBy(obj/item/I, mob/living/user)
 	var/list/tc = allowed_typecache
-	if(user.a_intent == INTENT_HARM || (I.flags_2 & HOLOGRAM_2) || (tc && !is_type_in_typecache(I, tc)))
+	if(user.a_intent == INTENT_HARM || (I.flags_2 & HOLOGRAM_2 | NO_MAT_REDEMPTION_2) || (tc && !is_type_in_typecache(I, tc)))
 		return FALSE
```

Comment:
@Cyberboss Actually, would it be cool if I added the chat message under here so it's not just a silent thing and people realize why nothing's happening when they click the machine?
```
if(!user.a_intent == INTENT_HARM)
    to_chat(user, "<span class='warning'>[parent] won't accept [I]!</span>")
```

Reply:
Not for the harm intent thing (that's supposed to hit X with Y). But, for the hologram and restrictions, yeah go ahead.